src/utils/monitoring-sites-data.js

```diff
@@ -111,6 +130,46 @@ const setRiverFlowSitesInBoundingBox = async (map, corners) => {
   map.getSource("riverFlowSites").setData(sites);
 };
 
+const setBristolWaterQualitySitesInBoundingBox = async (map) => {
+  const query = bristolWaterQualityQuery(map);
+  const response = await fetch(query).then((response) => response.json());
+  const records = response.records;
+  const siteIds = response.facet_groups[0].facets.map((facet) => facet.name);
+
+  const latestRecords = [];
+  for (const siteId of siteIds) {
+    let latestRecord = records.find(
+      (record) => record.fields.siteid === siteId
+    );
+
+    if (latestRecord) {
+      latestRecords.push(latestRecord);
+    } else {
+      const recordsForSiteQuery = getRecordsForBWQSiteQuery(siteId);
+      const response = await fetch(recordsForSiteQuery).then((response) =>
+        response.json()
+      );
+      const record = response.records[0];
+      latestRecords.push(record);
```

Comment:
If a record for a site isn't in the initial response (since they're ordered by datetime, there might be a site with only very old records), this is ensuring we get the latest record for that site.

Reply:
This is great, it might be worth adding this as a comment in the code itself to explain why we do this extra check, just so someone doesn't think it's redundant in the future.